### PR TITLE
Use 2.2 (2.0, 1.24 - not odd/dev versions) in check messages

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by phadej** [comment]
I'll fix tests soon-ish.

**Comment by phadej** [comment]
Fixed tests and found  few other odd version here and there.

**Comment by phadej** [comment]
@hvr, @23Skidoo is this good to merge?

**Comment by 23Skidoo** [comment]
Looks OK.

File: Cabal/Distribution/PackageDescription/Check.hs [hunk 1]
```diff
@@ -261,7 +261,7 @@ checkLibrary pkg lib =
   , checkVersion [1,25] (not (null (signatures lib))) $
       PackageDistInexcusable $
            "To use the 'signatures' field the package needs to specify "
-        ++ "at least 'cabal-version: >= 1.25'."
+        ++ "at least 'cabal-version: >= 2.0'."
```
Comment (hvr) [inline Cabal/Distribution/PackageDescription/Check.hs]: Can we please normalise to recommending `cabal-version: 2.0` w/o the `>=`? :-)

Comment (phadej) [reply Cabal/Distribution/PackageDescription/Check.hs]: Good point. Will do and merge after that.

**Pull request merged**